### PR TITLE
IOW-744 Switch to using MS Team email addresses for SNS notifications

### PR DESCRIPTION
**Pull request by kkehl-usgs**
Before making a pull request
----------------------------

- [x] Make sure all tests run
- [x] Update the changelog appropriately

Title
-----------
Brief description of changes. Reference the JIRA ticket if appropriate

Description
-----------
Created mailing lists inside secrets manager, which are managed by a lambda function that runs once a day.

After making a pull request
---------------------------
- [x] If appropriate, put the link to the PR in the JIRA ticket
- [ ] Assign someone to review unless the change is trivial

### CODE REVIEW

File: src/subscriptions_handler.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    response = client.list_subscriptions_by_topic(
+        TopicArn=topic_arn_warnings
+    )
+    logger.info(f"warning subscriptions: {response}")
+    subscribe_emails = secret_string['WARNINGS_LIST'].split(",")
+    logger.info(f"\nwarning subscribe emails: {subscribe_emails}")
+    _process_subscriptions(response, subscribe_emails, topic_arn_warnings)
+
+
+def _process_subscriptions(response, subscribe_emails, topic_arn):
```
Comment (ssoper-usgs) [inline src/subscriptions_handler.py:49]: this is neat!

**Review by ssoper-usgs — APPROVED** [review]

File: serverless.yml [hunk 2]
```diff
@@ -104,6 +104,24 @@ custom:
...
+    handler: src.subscriptions_handler.manage_subscriptions
+    role:
+      Fn::Sub:
+        - arn:aws:iam::${accountId}:role/csr-Lambda-Role
+        - accountId:
+            Ref: AWS::AccountId
+    environment:
+      AWS_DEPLOYMENT_REGION: ${self:provider.region}
+      STAGE: ${self:provider.stage}
+      LOG_LEVEL: INFO
```
Comment (ayan-usgs) [inline serverless.yml:117]: Are we sure we want the log level set to `INFO`.

File: src/subscriptions_handler.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+secrets_client = boto3.client('secretsmanager', os.getenv('AWS_DEPLOYMENT_REGION', 'us-west-2'))
+
+
+def manage_subscriptions(event, context):
+    """
+    Obtain a list of subscribers for terminal errors and a list of subscribers for warnings from the secrets
+    manager.  Process these lists to synchronize them with the subscriptions AWS knows about.
+    """
+    original = secrets_client.get_secret_value(
```
Comment (ayan-usgs) [inline src/subscriptions_handler.py:23]: I might have opted for the parameter store for this, but secrets manager is fine too.

File: src/subscriptions_handler.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    subscriptions = response['Subscriptions']
+    for subscription in subscriptions:
+        if subscription['Endpoint'] not in subscribe_emails and subscription['SubscriptionArn'] != "PendingConfirmation":
+            logger.info(f"\nunsubscribe {subscription['Endpoint']} because no longer on subscribe list")
+            logger.info(f"\nusing the SubscriptionArn {subscription}")
+            _unsubscribe_sns(subscription['SubscriptionArn'])
+
+
+def _subscribe_sns(event):
+    client = boto3.client('sns', os.getenv('AWS_DEPLOYMENT_REGION', 'us-west-2'))
```
Comment (ayan-usgs) [inline src/subscriptions_handler.py:71]: I noticed that `AWS_DEPLOYMENT_REGION` is being grabbed multiple times in this module. Consider grabbing it once at the beginning of the module or using a class to create a session from which clients can be spawned.

**Review by ayan-usgs — APPROVED** [review]

File: serverless.yml (hunk 2 above)
Comment (kkehl-usgs) [reply serverless.yml:117]: that was for debugging, I can remove ...

**Pull request merged**